framework/db/ActiveQuery.php

```diff
@@ -796,4 +789,89 @@ public function alias($alias)
         }
         return $this;
     }
+
+    /**
+     * Table names based on [[from]]
+     * @return string[] table names
+     * @throws \yii\base\InvalidConfigException
+     * @since 2.0.12
+     */
+    public function getTableNamesUsedInFrom()
+    {
```

Comment:
this method is the only one about getting tableNames. Shouldnt this get a cleaner name like 'getTableNames' or 'extractTableNames'